src/common/util/src/monitors/gpu_performance_counter.cpp

```diff
@@ -0,0 +1,235 @@
+                i += path.length() + 1;
+            } else {
+                break;
+            }
+        }
+        free(ExpandedPathList);
+        return pathList;
+    }
+
+    std::vector<PDH_HCOUNTER> addCounter(PDH_HQUERY Query, std::vector<std::string> pathList) {
+        PDH_STATUS Status;
+        std::vector<PDH_HCOUNTER> CounterList;
+        for (std::string path : pathList) {
+            PDH_HCOUNTER Counter;
+            Status = PdhAddCounterA(Query, path.c_str(), NULL, &Counter);
+            if (Status != ERROR_SUCCESS) {
+                return CounterList;
+            }
+            Status = PdhSetCounterScaleFactor(Counter, -2);  // scale counter to [0, 1]
+            if (ERROR_SUCCESS != Status) {
+                return CounterList;
+            }
+            CounterList.push_back(Counter);
+        }
+        return CounterList;
+    }
+
+    std::map<std::string, double> get_load() {
+        if (numDevices == 0)
+            return {{"00000000", 0}};
+        PDH_STATUS status;
+        auto ts = std::chrono::system_clock::now();
+        if (ts > lastTimeStamp) {
+            auto delta =
+                std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now() - lastTimeStamp);
+            if (delta.count() < 500) {
+                std::this_thread::sleep_for(std::chrono::milliseconds(500 - delta.count()));
+            }
+        }
+        lastTimeStamp = std::chrono::system_clock::now();
+        status = PdhCollectQueryData(query);
+        PDH_FMT_COUNTERVALUE displayValue;
+        std::map<std::string, double> gpuLoad;
+        for (auto item : coreTimeCounters) {
+            double value = 0;
+            auto coreCounters = item.second;
+            for (int counterIndex = 0; counterIndex < MAX_COUNTER_INDEX; counterIndex++) {
+                auto countersList = coreCounters[counterIndex];
+                for (auto counter : countersList) {
+                    status = PdhGetFormattedCounterValue(counter, PDH_FMT_DOUBLE, NULL, &displayValue);
+                    if (status != ERROR_SUCCESS) {
+                        continue;
+                    }
+                    value += displayValue.doubleValue;
+                }
+            }
+            gpuLoad[item.first] = value;
+        }
+        return gpuLoad;
+    }
+
+private:
+    QueryWrapper query;
+    std::map<std::string, std::vector<std::vector<PDH_HCOUNTER>>> coreTimeCounters;
+    std::chrono::time_point<std::chrono::system_clock> lastTimeStamp = std::chrono::system_clock::now();
+    std::size_t numDevices = 0;
+};
+
+#elif defined(__linux__)
+#    include <unistd.h>
+
+#    include <chrono>
+#    include <fstream>
+#    include <regex>
+#    include <utility>
+
+namespace ov {
+namespace util {
+namespace monitor {
+class GpuPerformanceCounter::PerformanceCounterImpl {
+public:
+    PerformanceCounterImpl() {}
+
+    std::map<std::string, double> get_load() {
+        // TODO: Implement.
```

Comment:
Linux OS will not be implement in this PR?

Reply:
Yes.